src/hardware/driver/mod.rs

```diff
@@ -123,11 +143,27 @@ impl LaserInterlock {
         }
     }
 
-    pub fn set(&mut self, reason: Option<Reason>) {
+    pub fn set(
+        &mut self,
+        reason: Option<Reason>,
+        output_state: &mut [output::sm::StateMachine<output::Output<I2c1Proxy>>;
+                 2],
+    ) -> [Option<fugit::MillisDuration<u64>>; 2] {
         // only update if no reason yet or if clearing (remember first reason)
         if self.reason.is_none() || reason.is_none() {
             self.pin.set_state(reason.is_none().into());
+            // disable both outputs if interlock is tripped
             self.reason = reason;
+            if self.reason.is_some() {
```

Comment:
A bit hard to parse mentally. Maybe clearer with a `match` or an inversion of thw two `if` s?